### PR TITLE
fontconfig: 2.13.93 -> 2.13.94

### PR DESCRIPTION
**Pull request by andresilva**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/fontconfig/default.nix [hunk 1]
```diff
@@ -15,21 +15,13 @@
...
-  version = "2.13.93";
+  version = "2.13.94";
 
   src = fetchurl {
     url = "http://fontconfig.org/release/${pname}-${version}.tar.xz";
-    sha256 = "1850q4k80yxma5g3yxkvyv8i5a3xqzswwml8gjy3jmywx8qqd5pa";
+    sha256 = "0g004r0bkkqz00mpm3svnnxn7d83158q0yb9ggxryizxfg5m5w55";
   };
 
-  patches = [
```
Comment (andresilva) [inline pkgs/development/libraries/fontconfig/default.nix:25]: No longer needed as it's included in the release.

**Review by jtojnar — APPROVED** [review]
[Changes](https://gitlab.freedesktop.org/fontconfig/fontconfig/-/compare/2.13.93...2.13.94) look okay.

Interesting commits:

- [Add support for XDG_DATA_DIRS](https://gitlab.freedesktop.org/fontconfig/fontconfig/-/commit/6f27f42e6140030715075aa3bd3e5cc9e2fdc6f1)
  Looks like it was prompted by Guix people. https://gitlab.freedesktop.org/fontconfig/fontconfig/-/issues/271

File: pkgs/development/libraries/fontconfig/default.nix [hunk 2]
```diff
@@ -15,21 +15,13 @@
 
 stdenv.mkDerivation rec {
   pname = "fontconfig";
-  version = "2.13.93";
+  version = "2.13.94";
 
   src = fetchurl {
     url = "http://fontconfig.org/release/${pname}-${version}.tar.xz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/fontconfig/default.nix:21]:
```suggestion
    url = "https://fontconfig.org/release/${pname}-${version}.tar.xz";
```

or the final redirect which would be ``https://www.freedesktop.org/software/fontconfig/release/``.

File: pkgs/development/libraries/fontconfig/default.nix [hunk 3]
```diff
@@ -15,21 +15,13 @@
...
-    sha256 = "1850q4k80yxma5g3yxkvyv8i5a3xqzswwml8gjy3jmywx8qqd5pa";
+    sha256 = "0g004r0bkkqz00mpm3svnnxn7d83158q0yb9ggxryizxfg5m5w55";
   };
 
-  patches = [
-    # Fix build with no docbook2html.
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/fontconfig/default.nix:27]: Please remove the unused fetchpatch from inputs.

**Comment by andresilva** [comment]
@SuperSandro2000 addressed

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129675` at 824c83d7 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>107 packages marked as broken and skipped:</summary>
  <ul>
    <li>adobe-reader</li>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-openj9-bin-13</li>
    <li>adoptopenjdk-jre-openj9-bin-14</li>
    <li>adoptopenjdk-openj9-bin-13</li>
    <li>adoptopenjdk-openj9-bin-14</li>
    <li>airtame</li>
    <li>autotrace</li>
    <li>blastem</li>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>clickshare-csc1</li>
    <li>curaByDagoma</li>
    <li>cura_stable</li>
    <li>displaycal</li>
    <li>dvd-slideshow</li>
    <li>eagle7</li>
    <li>electron_3</li>
    <li>electron_4</li>
    <li>electron_5</li>
    <li>epsxe</li>
    <li>erlang_basho_R16B02</li>
    <li>eterm</li>
    <li>evilvte</li>
    <li>ffmpeg_2</li>
    <li>ffmpeg_2_8</li>
    <li>ffmpeg_3</li>
    <li>ffmpeg_3_4</li>
    <li>globalplatform</li>
    <li>gnash</li>
    <li>gnuradio3_7</li>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gnuradio</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.limesdr</li>
    <li>gnuradio3_7Packages.nacl</li>
    <li>gnuradio3_7Packages.osmosdr</li>
    <li>gnuradio3_7Packages.rds</li>
    <li>googleearth-pro</li>
    <li>gppcscconnectionplugin</li>
    <li>gpshell</li>
    <li>grass</li>
    <li>green-pdfviewer</li>
    <li>gtmess</li>
    <li>hadoop</li>
    <li>hadoop_2_7</li>
    <li>hadoop_2_8</li>
    <li>hadoop_2_9</li>
    <li>hplipWithPlugin_3_16_11</li>
    <li>hplip_3_16_11</li>
    <li>impressive</li>
    <li>invoice2data</li>
    <li>jboss</li>
    <li>jfbpdf</li>
    <li>jfbview</li>
    <li>k2pdfopt</li>
    <li>keyfinder</li>
    <li>libav</li>
    <li>libav_0_8</li>
    <li>libav_11</li>
    <li>libav_12</li>
    <li>libgroove</li>
    <li>linuxPackages_hardkernel_4_14.akvcam</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_4_14.xpadneo</li>
    <li>linuxPackages_hardkernel_latest.akvcam</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.xpadneo</li>
    <li>loxodo</li>
    <li>metamorphose2</li>
    <li>mfcl8690cdwcupswrapper</li>
    <li>mrxvt</li>
    <li>mupdf_1_17</li>
    <li>natron</li>
    <li>obinskit</li>
    <li>ocropus</li>
    <li>odpdown</li>
    <li>openrw</li>
    <li>opentsdb</li>
    <li>oven-media-engine</li>
    <li>pcsxr</li>
    <li>pdfdiff</li>
    <li>pdfread</li>
    <li>pgadmin</li>
    <li>pig</li>
    <li>plover.stable</li>
    <li>riak</li>
    <li>ring-daemon</li>
    <li>scallion</li>
    <li>scribus</li>
    <li>scyther</li>
    <li>spark</li>
    <li>staruml</li>
    <li>stepmania</li>
    <li>termplay</li>
    <li>torchat</li>
    <li>tvheadend</li>
    <li>twinkle</li>
    <li>unicap</li>
    <li>vdrPlugins.markad</li>
    <li>vdrPlugins.vaapidevice</li>
    <li>vdrPlugins.xineliboutput</li>
    <li>winpdb</li>
    <li>xpdf</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/666b7446759a0d4b07436697b2aadd30">ocz-ssd-guru</a></li>
  </ul>
</details>
<details>
  <summary>15532 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>CHOWTapeModel</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>EmptyEpsilon</li>
    <li>MIDIVisualizer</li>
    <li>OSCAR</li>
    <li>OVMF</li>
    <li>OVMF-CSM</li>
    <li>OVMF-secureBoot</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>25 packages built successfully:</summary>
  <ul>
    <li>atk (gnome.atk ,gnome2.atk)</li>
    <li>cairo (gnome2.cairo)</li>
    <li>fontconfig</li>
    <li>gd</li>
    <li>gdk-pixbuf</li>
    <li>gobject-introspection (gnome.gobject-introspection)</li>
    <li>gsettings-desktop-schemas (gnome.gsettings-desktop-schemas)</li>
    <li>libsoup (gnome.libsoup)</li>
    <li>pango (gnome2.pango)</li>
    <li>harfbuzz</li>
    <li>json-glib</li>
    <li>lib2geom</li>
    <li>libass</li>
    <li>libnotify</li>
    <li>libva</li>
    <li>libva-minimal</li>
    <li>libxkbcommon (libxkbcommon_8)</li>
    <li>python39Packages.pycairo</li>
    <li>python39Packages.pygobject3</li>
    <li>spideroak</li>
    <li>wayland-protocols</li>
    <li>wayland-scanner</li>
    <li>xfractint</li>
    <li>xlibsWrapper (xorg.xlibsWrapper)</li>
    <li>xorg.libXft</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `substituteAll`.
Near pkgs/development/libraries/fontconfig/default.nix:2:3:
```
  |
2 | , substituteAll
  |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by das-g** [comment]
Is there a way to fast-pace a mass-rebuild-inducing change like this in order to unbreak what's currently broken on `master` and on the following three channels?
- ` nixpkgs-unstable`
- ` nixos-unstable-small`
- ` nixos-unstable`

I guess most affected applications _can_ be used despite the current broken font rendering behavior, but it sure isn't a pleasant experience. (Speaking as someone affected by #129733) Sure, one can work around this by running the application from a not-affected nixpkgs-revision despite the system being on an affected revision, or one can reset one's whole system to such a revision (or to some not affected channel), but I don't think users should be expected to have to do that any longer than the shortest periods of time.

**Comment by andresilva** [comment]
I'm not the best person to answer this since I'm just a random NixOS contributor. I asked in the current staging PR update to include this (https://github.com/NixOS/nixpkgs/pull/129506#issuecomment-876623033), other than waiting for that to happen and then waiting for staging to get merged I'm not sure what we can do. Personally I just went with sticking to an old nixpkgs-revision and not updating which is not a great solution :(

**Comment by vcunat** [comment]
The `staging-next` branch has almost all binaries for Linuxes, so you could use it already.  It might even get merged over this weekend, even though *-darwin certainly still isn't in ideal shape.

**Pull request merged**